tempfork/spf13/cobra/comp.zsh

```diff
@@ -0,0 +1,198 @@
+# Copyright 2013-2023 The Cobra Authors
```

Comment:
I'm having trouble getting this to work with my zsh setup.  I'm not following the `tailscale completion` directions exactly, so it may be unique to me.  I'm going to keep hacking away at it.

But one thing I came across is that https://zsh.sourceforge.io/Doc/Release/Completion-System.html mentions:

> When compinit is run, it searches all such files accessible via fpath/FPATH and reads the first line of each of them. This line should contain one of the tags described below. Files whose first line does not start with one of these tags are not considered to be part of the completion system and will not be treated specially.

So I think we need the `#compdef` line before the license header.  Changing that locally doesn't fix my problem, so I've got something else going on, but I think this change is probably necessary anyway.

Reply:
Hey @willnorris, oh great shout! I've only been testing this locally by sourcing it into my active shell like `. <( tailscale completion zsh )`. That should still work. I'll have a test of loading the rest from disk.

Reply:
yeah, loading it directly into the current shell like that works.  It's having it autoload on a new shell that I'm having trouble with.

Reply:
okay, actually that did fix it for me.  Moving the `#compdef` line to the top of the file, then running `compinit` again seems to work.  (I was missing running `compinit` before, so it wasn't obvious that this did fix it)

Reply:
> So I think we need the `#compdef` line before the license header.

Thanks for this pointer, @willnorris. That did indeed fix it. I've now tested those instructions for bash, zsh, fish and powershell too and I believe they are all working. Thank you for taking it for a spin <3